multimedia/Sunshine/Portfile

```diff
@@ -0,0 +1,75 @@
+platforms                darwin
+license                  GPL-3
+maintainers              {@SunshineStream sunshinestream}
+description              Sunshine is a Gamestream host for Moonlight.
+long_description         {*}${description}
+homepage                 https://sunshinestream.github.io
+master_sites             https://github.com/sunshinestream/sunshine/releases
+
+fetch.type               git
+post-fetch {
+    system -W ${worksrcpath} "${git.cmd} submodule update --init --recursive"
+}
+
+depends_lib              port:avahi \
+                         port:ffmpeg \
+                         port:libopus
+
+boost.version            1.76
+
+configure.args           -DBOOST_ROOT=[boost::install_area] \
+                         -DSUNSHINE_ASSETS_DIR=${prefix}/etc/sunshine/assets \
+                         -DSUNSHINE_CONFIG_DIR=${prefix}/etc/sunshine/config
+
+cmake.out_of_source      yes
+
+startupitem.create       yes
+startupitem.executable   "${prefix}/bin/{$name}"
+startupitem.location     LaunchDaemons
+startupitem.name         ${name}
+startupitem.netchange    yes
+
+# is this actually necessary? this should all be handled by CMakeLists
+destroot {
+    # install assets
+    xinstall -d -m 755 ${destroot}${prefix}/etc/${name}/assets
+    xinstall {*}[glob ${worksrcpath}/src_assets/common/assets/*.*] ${destroot}${prefix}/etc/${name}/assets
+    xinstall {*}[glob ${worksrcpath}/src_assets/macos/assets/*.*] ${destroot}${prefix}/etc/${name}/assets
+
+    # install web assets
+    xinstall -d -m 755 ${destroot}${prefix}/etc/${name}/assets/web/fonts/fontawesome-free-web/css
+    xinstall -d -m 755 ${destroot}${prefix}/etc/${name}/assets/web/fonts/fontawesome-free-web/webfonts
+    xinstall {*}[glob ${worksrcpath}/src_assets/common/assets/web/*.*] ${destroot}${prefix}/etc/${name}/assets/web
+    xinstall {*}[glob ${worksrcpath}/src_assets/common/assets/web/fonts/fontawesome-free-web/*.*] ${destroot}${prefix}/etc/${name}/assets/web/fonts/fontawesome-free-web
+    xinstall {*}[glob ${worksrcpath}/src_assets/common/assets/web/fonts/fontawesome-free-web/css/*.*] ${destroot}${prefix}/etc/${name}/assets/web/fonts/fontawesome-free-web/css
+    xinstall {*}[glob ${worksrcpath}/src_assets/common/assets/web/fonts/fontawesome-free-web/webfonts/*.*] ${destroot}${prefix}/etc/${name}/assets/web/fonts/fontawesome-free-web/webfonts
+    xinstall -d -m 755 ${destroot}${prefix}/etc/${name}/assets/web/images
+    xinstall -d -m 755 ${destroot}${prefix}/etc/${name}/assets/web/third_party
+    xinstall {*}[glob ${worksrcpath}/src_assets/common/assets/web/images/*.*] ${destroot}${prefix}/etc/${name}/assets/web/images
+    xinstall {*}[glob ${worksrcpath}/src_assets/common/assets/web/third_party/*.*] ${destroot}${prefix}/etc/${name}/assets/web/third_party
+
+    xinstall -d -m 755 ${destroot}${prefix}/etc/${name}/config
+
+    # install sunshine.conf
+    xinstall {*}[glob ${worksrcpath}/src_assets/common/config/*.*] ${destroot}${prefix}/etc/${name}/config
```

Comment:
you probably don't want to overwrite a configuration file if someone would update to a newer version. I'm sure there are many Portfiles that take care of things like that and/or there might be a description in the guide/Wiki on how to handle this. Perhaps the `phpmyadmin` is one example where you can see that it checks if a configuration file is already present and depending on the outcome does a different action.

Reply:
I tried to find examples but couldn't find one.

The documentation is not complete. https://guide.macports.org/#development.practices.dont-overwrite

Reply:
I suggested a port that does similar things to check how one could do this (i.e., `phpadmin`); another one that uses `glob` is the `cppcheck` port. 

I agree that some parts of the guide are incomplete; sometimes you can find that information in the Wiki. For example, in this case you could check [here](https://trac.macports.org/wiki/PortfileRecipes#configfiles).

Reply:
Thank you, sorry I must have read it too fast previously... The link you provided is good though!

I'll use these as references:
https://trac.macports.org/browser/trunk/dports/security/stegdetect/Portfile#L49
https://trac.macports.org/browser/trunk/dports/sysutils/bacula/Portfile#L210

Reply:
I'm sure I did something wrong, but tested this and it doesn't appear the `post-activate` works with a `pkg` generated using `sudo port pkg sunshine`

```txt
# Rename files in `destroot`
post-destroot {
    file rename ${destroot}${prefix}/etc/${name}/config/sunshine.conf ${destroot}${prefix}/etc/${name}/config/sunshine.conf.sample
    file rename ${destroot}${prefix}/etc/${name}/config/apps.json ${destroot}${prefix}/etc/${name}/config/apps.json.sample
}

# Don't overwrite existing preference files
post-activate {
    if {![file exists ${prefix}/etc/${name}/config/apps.json]} {
        file copy ${destroot}${prefix}/etc/${name}/config/apps.json.sample \
            ${prefix}/etc/${name}/config/apps.json
        }
    if {![file exists ${prefix}/etc/${name}/config/sunshine.conf]} {
        file copy ${destroot}${prefix}/etc/${name}/config/sunshine.conf.sample \
            ${prefix}/etc/${name}/config/sunshine.conf
        }
}
```

```txt
reenignearcher@ReenigneArchers-iMac-Pro MacOS % ls /opt/local/etc/Sunshine/config
apps.json.mp_1655322422		sunshine.conf.mp_1655322422
apps.json.sample		sunshine.conf.sample
```

It does work when installing using the Portfile. Is there a way to do this that will work for the `pkg` as well?